### PR TITLE
Teach `-Z treat-err-as-bug` to take a number of errors to emit

### LINKED ISSUE #58983: Make `-Z treat-err-as-bug` accept a number of errors to skip
Allow `-Z treat-err-as-bug` to take a number, in order to make it easier to debug errors that can't be made to be emitted _first_. In my mind, you'd call `rustc -Z treat-err-as-bug 2` on a file with three errors and the first two would be emitted, you'd get the backtrace for the second one, and the third one wouldn't be emitted.

### PR DESCRIPTION
**Pull request by estebank**
`-Z treat-err-as-bug` will cause `rustc` to panic after the first error is reported, like previously. `-Z treat-err-as-bug=2` will cause `rustc` to panic after 2 errors have been reported.

Fix #58983.

### CODE REVIEW

**Comment by estebank** [comment]
CC @rust-lang/compiler

File: src/librustc/session/config.rs [hunk 1]
```diff
@@ -3212,7 +3212,7 @@ mod tests {
         assert!(reference.dep_tracking_hash() != opts.dep_tracking_hash());
 
         opts = reference.clone();
-        opts.debugging_opts.treat_err_as_bug = true;
+        opts.debugging_opts.treat_err_as_bug = Some(1);
```
Comment (estebank) [inline src/librustc/session/config.rs]:
```suggestion
        opts.debugging_opts.treat_err_as_bug = Some(0);
```

**Comment by oli-obk** [comment]
r=me with Travis passing.

@bors r=oli-obk

**Comment by oli-obk** [comment]
Oops.

@bors r-

r? oli-obk

**Pull request merged**